src/main/java/de/caritas/cob/uploadservice/api/facade/UploadFacade.java

```
@@ -118,7 +132,14 @@ private void sanitizeAndEncryptParametersAndUploadToRocketChatRoom(
       throw new InternalServerErrorException(e, LogService::logEncryptionServiceError);
     }
 
-    rocketChatService.roomsUpload(rocketChatCredentials, encryptedRocketChatUploadParameter);
+    FullUploadResponseDto uploadResponse = rocketChatService.roomsUpload(rocketChatCredentials,
+        encryptedRocketChatUploadParameter);
+
+    if (doAttachmentE2e) {
+      assert uploadResponse.getMessage() != null;
+      mongoDbService.setE2eType(uploadResponse.getMessage().getId());
```

If setting this fails, the frontend gets no feedback. Can you throw an error in that case, please?

---

Done 👍